### ISSUE TITLE
Audio Router cannot route audio from Overwatch

### ISSUE THREAD

**Issue by dtphase** [wolfreak99/audio-router#8]
Hi wolfreak99! 

Thank you so much for picking up this project, it really gives me hope for the future. 0.10.2 is working fine for me but I cannot route or duplicate the program Overwatch.exe's sound (presumably due to anticheat).

When I do try it returns the error: "Error code 0x0000007E: The specified module could not be found. Router functionality not available."

I probably wouldn't have mentioned it except that Chevolume has no problem with Overwatch.

Also for me Duplicate and Route both seem to just Route. This should probably be a separate issue but I have a feeling it's already been reported.

**Comment by wolfreak99** [comment]
Hello there, @dtphase. First off, thank you for taking your time to try out my fork of the program and to also report your issues. I know there are some programs that won't work, in fact i've seen some stuff about the original audiorouter code that some people have said it's still very buggy so, for being one of my first few c++ projects, I feel i'm throwing myself into a bit of a challenge with this one, and I honestly didn't intend to get this fork of it much attention like it seems to have gotten because I wasn't ready for this much yet (I'm still going to try to pull through and try my best however).

I personally do not have overwatch to test this myself, and I'm still in the process of going through and trying to reorganize the code into a more maintainable state, and add some features that will allow easier debugging down the line (such verbose logging information). The code that I forked from @audiorouterdev was pretty chaotic and I'm still trying to make the gist of what all is going on here.

I believe I know what you're talking about with the duplicate and route both doing the same, unfortunately I've not yet studied the code enough to fix this just yet. Would you be able to make a seperate issue for the Duplicate and Route both seeming to just route, that way I can be able to keep track of these easier? 

I'm also about to do a commit where I've replaced "Router Functionality not available." with "Router functionality not available. (source file/function name)". Once I push it, I will comment and ask if you can update the code and try to replicate the overwatch issue and tell me where the error is coming from

**Comment by dtphase** [comment]
Hi again Wolf, I created a new issue for the route/duplicate issue. Have you committed the changes you were talking about yet? I cloned and built the program but had no changes in the error.

**Comment by wolfreak99** [comment]
@dtphase , I have just created a pull request which adds a stacktrace to the errors at #11  Would you be able to clone that pr and see what the stacktrace says? Are you using visual studio to build the program?

**Comment by dtphase** [comment]
Exception thrown at 0x00007FF810A39E08 in Audio Router.exe: Microsoft C++ exception: stacktrace::stack_exception<std::runtime_error> at memory location 0x00000064F853DE70.
Exception thrown at 0x00007FF810A39E08 in Audio Router.exe: Microsoft C++ exception: std::exception at memory location 0x00000064F853EC28.


I am using Visual Studio, yes. I'm very new at this though so I hope this is what you were asking for. There was nothing in the call stack window and this was the closest to anything helpful I could find.

**Comment by wolfreak99** [comment]
@dtphase  No, but likely I broke something along the way of implementing it (which is why i use pull requests instead of pushing code directly to the commit, just a tip if you're new and starting with git).

It would actually help if i was able to debug this via a remote session. Would you happen to have Teamviewer and facebook by chance? if so, send me a message here: https://www.facebook.com/jbminor1991

**Comment by EntityReborn** [comment]
I am also seeing similar issues with Spotify. I am using the latest version of Spotify (installed, not the Store App), and the latest of Audio Router from your Appveyor. Windows 10 Pro 1803 x64.
https://i.imgur.com/GVl6JYv.png

**Comment by dedmen** [comment]
Had the same above issue with spotify, but latest develop branch build (built by myself) works fine.
